oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java

```diff
@@ -658,31 +680,85 @@ private APIGatewayProxyResponseEvent handleAuthJourney(
+                new OrchSessionItem(previousSession)
                         .withSessionId(newSessionId)
-                        .withTimeToLive(
-                                NowHelper.nowPlus(
-                                                configurationService.getSessionExpiry(),
-                                                ChronoUnit.SECONDS)
-                                        .toInstant()
-                                        .getEpochSecond());
-        orchSessionService.addSession(updatedOrchSession);
-        orchSessionService.deleteSession(existingOrchSession.getSessionId());
+                        .withTimeToLive(timeNow + configurationService.getSessionExpiry());
+        orchSessionService.addSession(updatedSession);
+        orchSessionService.deleteSession(previousSession.getSessionId());
         LOG.info(
                 "Updated existing Orch session ID from {} to {}",
-                existingOrchSession.getSessionId(),
+                previousSession.getSessionId(),
                 newSessionId);
-        return updatedOrchSession;
+        return updatedSession;
     }
 
-    private OrchSessionItem createNewOrchSession(String sessionId) {
-        var newOrchSessionItem = new OrchSessionItem(sessionId);
-        orchSessionService.addSession(newOrchSessionItem);
-        LOG.info("Created new Orch session with session ID: {}", sessionId);
-        return newOrchSessionItem;
+    private OrchSessionItem updateOrchSessionDueToMaxAgeExpiry(
+            String newSessionId, OrchSessionItem previousSession) {
+        String newSessionIdForPreviousSession = IdGenerator.generate();
+        OrchSessionItem updatedPreviousSession =
+                new OrchSessionItem(previousSession)
+                        .withSessionId(newSessionIdForPreviousSession)
+                        .withTimeToLive(timeNow + configurationService.getSessionExpiry());
+        orchSessionService.addSession(updatedPreviousSession);
+        orchSessionService.deleteSession(previousSession.getSessionId());
+        LOG.info(
+                "Updated previous Orch session due to max age expiry. Session ID updated from {} to {}",
+                previousSession.getSessionId(),
+                newSessionIdForPreviousSession);
+
+        OrchSessionItem newSession =
+                new OrchSessionItem(previousSession)
+                        .withSessionId(newSessionId)
+                        .withTimeToLive(timeNow + configurationService.getSessionExpiry())
+                        .withAuthenticated(false)
+                        .withPreviousSessionId(newSessionIdForPreviousSession);
+        orchSessionService.addSession(newSession);
+        LOG.info("Created new Orch session with session ID {} due to max age expiry", newSessionId);
+        return newSession;
+    }
+
+    private Optional<Long> getMaxAge(AuthenticationRequest authRequest) {
+        Optional<Long> maxAge = Optional.empty();
+        try {
```

Comment:
I wonder if we want to be pulling these methods out this file, as it's getting so long. I created an `AuthorisationHelper` file a little while back - might be nice to move them into there?

Reply:
As discussed, will add this to tech debt in order to get this deployed and unblock other max age work